awx/main/utils/common.py

```diff
@@ -1079,3 +1082,47 @@ def truncate_stdout(stdout, size):
             set_count += 1
 
     return stdout + u'\u001b[0m' * (set_count - reset_count)
+
+def create_partition(start=None, end=None, partition_label=None, minutely=True):
+    """Creates new partition tables for events.
+    - start defaults to beginning of current hour
+    - end defaults to end of current hour
+    - partition_label defaults to YYYYMMDD_HH
+
+    - minutely will create partitions that span _a single minute_ for testing purposes
+    """
+    current_time = now()
+    if not start:
+        if minutely:
+            start = current_time.replace(microsecond=0, second=0)
+        else:
+            start = current_time.replace(microsecond=0, second=0, minute=0)
+    if not end:
+        if minutely:
+            end = start.replace(microsecond=0, second=0) + timedelta(minutes=1)
+        else:
+            end = start.replace(microsecond=0, second=0, minute=0) + timedelta(hours=1)
+    start_timestamp = str(start)
+    end_timestamp = str(end)
+
+    if not partition_label:
+        if minutely:
+            partition_label = start.strftime('%Y%m%d_%H%M')
+        else:
+            partition_label = start.strftime('%Y%m%d_%H')
+
+    with connection.cursor() as cursor:
+        # Only partitioning main_jobevent on first pass
```

Comment:
Similar to the "make it work for all use cases" comment above, you could probably lose this loop, and just pass `tblname` _directly into_ this function so that the proper parent table is operated on depending on the context from which this function is called (either in the upgrade-time migration, or via the task manage as new jobs are scheduled).